modules/component-store/src/component-store.ts

```diff
@@ -225,40 +228,71 @@ export class ComponentStore<T extends object> implements OnDestroy {
     projector: (s: T) => Result,
     config?: SelectConfig
   ): Observable<Result>;
+  select<SelectorsObject extends Record<string, ObservableInput<unknown>>>(
+    selectorsObject: SelectorsObject,
+    config?: SelectConfig
+  ): Observable<{
+    [K in keyof SelectorsObject]: ObservedValueOf<SelectorsObject[K]>;
+  }>;
   select<Selectors extends Observable<unknown>[], Result>(
-    ...args: [...selectors: Selectors, projector: Projector<Selectors, Result>]
+    ...selectorsWithProjector: [
+      ...selectors: Selectors,
+      projector: Projector<Selectors, Result>
+    ]
   ): Observable<Result>;
   select<Selectors extends Observable<unknown>[], Result>(
-    ...args: [
+    ...selectorsWithProjectorAndConfig: [
       ...selectors: Selectors,
       projector: Projector<Selectors, Result>,
       config: SelectConfig
     ]
   ): Observable<Result>;
   select<
-    Selectors extends Array<Observable<unknown> | SelectConfig | ProjectorFn>,
+    Selectors extends Array<
+      Observable<unknown> | SelectConfig | ProjectorFn | SelectorsObject
+    >,
     Result,
-    ProjectorFn extends (...a: unknown[]) => Result
+    ProjectorFn extends (...a: unknown[]) => Result,
+    SelectorsObject extends Record<string, ObservableInput<unknown>>
   >(...args: Selectors): Observable<Result> {
-    const { observables, projector, config } = processSelectorArgs<
-      Selectors,
-      Result,
-      ProjectorFn
-    >(args);
+    const { observablesOrSelectorsObject, projector, config } =
+      processSelectorArgs<Selectors, Result, ProjectorFn, SelectorsObject>(
+        args
+      );
 
     let observable$: Observable<Result>;
-    // If there are no Observables to combine, then we'll just map the value.
-    if (observables.length === 0) {
-      observable$ = this.stateSubject$.pipe(
-        config.debounce ? debounceSync() : (source$) => source$,
-        map((state) => projector(state))
+
+    // Operators to apply to the pipe.
+    // Many 'any's because we are pushing operators manually into an array
+    // before passing them into a pipe.
+    const operators: OperatorFunction<any, unknown>[] = [];
+
+    if (config.debounce) {
+      operators.push(debounceSync());
+    }
+
+    if (projector) {
+      // where projector is provided.
+      operators.push(
+        map((projectorArgs: unknown[] | object): Result => {
+          if (Array.isArray(projectorArgs)) {
+            return projector(...projectorArgs);
+          }
+          return projector(projectorArgs);
+        })
       );
+    }
+
+    if (
+      Array.isArray(observablesOrSelectorsObject) &&
+      observablesOrSelectorsObject.length === 0
+    ) {
+      // Just a projector that could have debounced.
+      observable$ = this.stateSubject$.pipe(...(operators as [any]));
```

Comment:
Just for clarity, why use `[any]` here instead of `any[]` for consistency of the casting?

Reply:
Removed and kept the pipe operators inline.